### DISCUSSION TITLE
question about keep-alive

### DISCUSSION THREAD

**Discussion by imba-tjd** [hatoo/oha#600] [General]
Hi. As far as I know, browser uses the Keep-Alive mechanism by default for HTTP/1.1. Why do you say *In real, user doesn't query same URL using Keep-Alive*?

![图片](https://github.com/user-attachments/assets/5cc1a245-27ac-4fdd-928d-3ca7b3f6fefe)

![图片](https://github.com/user-attachments/assets/842efcd4-2b33-4766-9acb-4d27b2a1cf66)

**Comment by hatoo** [comment]
I assume `oha` is used to benchmark a single URL (although `-rand-regex-url` feature is implemented)
And by disabling `Keep-Alive`, we can mimic a situation where many browsers access the same URL.

Benchmarking a single URL with `Keep-Alive`  is like a
```
GET / HTTP/1.1
...
GET / HTTP/1.1 # in the same connection, it is unusual since you must know the content of it already
...
GET / HTTP/1.1
...
```
It is something weird.

**Reply by imba-tjd** [reply]
Thanks. I think i know. I just read the article about "coordinated-omission"

The goal for benchmark tool is to simulate many users connecting at the same time, rather than a single user sending many request. Disable Keep-Alive can achieve this.

~~But I think the wording in readme needs changing.~~ Or maybe when someone else has this question, give them a link to this this discussion.
